lib/remote/apiuser.cpp

```diff
@@ -34,3 +50,47 @@ ApiUser::Ptr ApiUser::GetByClientCN(const String& cn)
 
 	return nullptr;
 }
+
+bool ApiUser::ComparePassword(String password) const {
+	String other_passwd = CreateHashedPasswordString(password, this->m_Salt, false);
+
+	const char *p1 = other_passwd.CStr();
+	const char *p2 = this->m_Hashed_passwd.CStr();
+
+	volatile char c = 0;
+
+	for (size_t i=0; i<64; ++i)
+		c |= p1[i] ^ p2[i];
+
+	return (c == 0);
+}
+
+Dictionary::Ptr ApiUser::GetPasswordDict(void)
+{
+	String passwd = this->GetPasswordHash();
+	if (passwd.IsEmpty() || passwd[0] != '$')
+		return NULL;
+
+	String::SizeType salt_begin = passwd.FindFirstOf('$', 1);
+	String::SizeType passwd_begin = passwd.FindFirstOf('$', salt_begin+1);
```

Comment:
Minor style: Add spaces between variable + 1. Also rename variables to saltBegin and passwordBegin.